### PR TITLE
Edit Group Chat

### PR DESCRIPTION
**Pull request by imphungky**
## DEV-004

Fixes: #

- [X] Has the code been formatted?
- [ ] Have dependencies/packages been added/removed?
      If so, please list them.

### Changes

List out all the changes, and then explain the changes made at a higher level

- Modified chat/[id].jsx to contain the id and check user permissions for editting
- created a new form for editing containing current chat info
- modified query to fetch for more group chat information for editing

### CODE REVIEW

File: client/content/locale/en.json [hunk 1]
```diff
@@ -44,5 +44,8 @@
   "find-groupchats": "FIND GROUPCHATS",
   "details": "Details",
   "created": "Created",
-  "last-modified": "Last modified"
+  "last-modified": "Last modified",
```
Comment (jcserv) [inline client/content/locale/en.json:47]: Could you also add these to fr.json?

File: client/pages/chat/[id].jsx [hunk 2]
```diff
@@ -1,19 +1,43 @@
...
+import React, { useEffect, useState } from "react";
 
 import client from "../../apollo-client";
 import ChatInfo from "../../components/ChatInfo";
 import { localesArr } from "../../content/locale";
 import { GET_GROUPCHAT, GET_GROUPCHAT_IDS } from "../../gql/GroupChat";
+import { GET_USER } from "../../gql/User";
+
+export default function Chat({ chat, id }) {
+  const [editPermissions, setEditPermissions] = useState(false);
```
Comment (jcserv) [inline client/pages/chat/[id].jsx]: Nice! Do you think we could abstract this into something like a permission wrapper component?
I have no clue tbh I'm just wondering if we could apply this for other features

File: server/resolvers/GroupChatResolver.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { GroupChat } from "../models";
 import { createGroupChatInput } from "../inputs";
 import { GroupChatIds, GroupChatPaginiated } from "../models/Groupchat";
 import { escapeRegex } from "../helpers";
+import { courseInformationInput } from "../inputs/GroupChatInput";
```
Comment (jcserv) [inline server/resolvers/GroupChatResolver.ts]: unused

File: server/resolvers/GroupChatResolver.ts [hunk 4]
```diff
@@ -132,13 +133,53 @@ export class GroupChatResolver {
...
+    const groupChat = await GroupChatModel.findOne({ _id: id });
+		if (!groupChat) {
+			return null;
+		}
+		groupChat.status = status;
+		const result = await groupChat.save();
+		return result;
+  }
+
+  @Mutation(returns => Boolean)
```
Comment (jcserv) [inline server/resolvers/GroupChatResolver.ts]: I think you can change returns to () to remove the eslint warning

File: client/components/DeleteChatModal.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    if (deleteResult) {
+      toast({
+        title: "Success",
+        description: `Chat has been deleted`,
+        status: "success",
+        position: "bottom-left",
+        duration: 5000,
+        isClosable: false,
+      });
+    }
```
Comment (jcserv) [inline client/components/DeleteChatModal.jsx:69]: can we get a toast for when the delete is unsuccessful as well?

File: client/components/DeleteChatModal.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      preserveScrollBarGap
+      isCentered
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>{formatMessage(messages.deleteMessage)}</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody alignContent="center">
+          <Button colorScheme="red" onClick={deleteAction}>
+            {formatMessage(messages.confirm)}
```
Comment (jcserv) [inline client/components/DeleteChatModal.jsx:87]: currently this content isn't populating for me

**Review by jcserv — CHANGES_REQUESTED** [review]
Great work, just a bit of feedback but after that I think it's good to go - pending a review from @AipioxTechson

**Review by AipioxTechson — COMMENTED** [review]
Aside from Jerry's requested changes, it looks good!

File: client/components/DeleteChatModal.jsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    });
+    if (deleteResult) {
+      toast({
+        title: "Success",
+        description: `Chat has been deleted`,
+        status: "success",
+        position: "bottom-left",
+        duration: 5000,
+        isClosable: false,
+      });
```
Comment (jcserv) [inline client/components/DeleteChatModal.jsx:57]: Can you redirect the user to the homepage after deleting?

**Pull request merged**